src/pages/admin/student/edit/page.tsx

```diff
@@ -1,17 +1,39 @@
 import * as React from 'react';
-import {Link} from 'react-router';
-import {adminRouteEnums} from '../../../../common/routeEnums/admin';
+import { Link } from 'react-router';
+import { Student } from '../../../../model/student/student';
+import { IEditStudentErrors } from '../../../../model/student/editStudentErrors';
+import { EditStudentComponent } from './components/editStudentComponent';
+const classNames: any = require('./pageStyles.scss');
 
-export class EditStudentPage extends React.Component<{}, {}> {
-   public render() {
-       return (
-         <div>
-           <span> Edit Student Page: </span>
-           <br/>
-           <br/>
-           <Link to={adminRouteEnums.student.list}>Back to student list</Link>
-           <Link to={adminRouteEnums.default}>Back to Dashboard</Link>
-         </div>
-        );
+interface IProps extends React.Props<EditStudentPage> {
+  studentId: number;
+  student: Student;
+  editStudentError: IEditStudentErrors;
+  fetchStudent: (id: number) => void;
+  updateStudent: (viewModel: Student, fieldName: string, value: string) => void;
+  saveStudent: (student: Student) => void;
+}
+
+export class EditStudentPage extends React.Component<IProps, {}> {
+  public componentDidMount() {
```

Comment:
componentDidMount is not fired when accessing EditStudentPage from a react-route Link component.
I've been digging a bit and seems a problem with Link.

Should I use componentWillReceiveProps instead to get the id of the student to fetch?